### PR TITLE
hdf5back: fix use of deallocated string mem. Fixes #1232

### LINKED ISSUE #1232: HDF5 unit test failures with gcc 5.x and clang 3.7
Here is a link to the failures: https://gist.github.com/rwcarlsen/3c35df49b8db6a7de0b5

Everything compiles fine, but the failures occur consistently at runtime.  They are observable on Arch, Ubuntu, and Mac with the newer versions of both gcc and clang.  Not sure if it's HDF5's problem or a problem with the way we are using HDF5.  They are deterministic and occur every time.  Integration tests that use HDF5 also fail, and I can't successfully run any simulation with the HDF5 backend with the newer compilers.

@scopatz - any ideas?

[edit] updated gist link

### PR DESCRIPTION
**Pull request by rwcarlsen**

### CODE REVIEW

**Comment by Baaaaam** [comment]
That solve a lot of issue on Mac OS !!!
>  [----------] Global test environment tear-down
> [==========] 553 tests from 109 test cases ran. (13216 ms total)
> [  PASSED  ] 552 tests.
> [  FAILED  ] 1 test, listed below:
> [  FAILED  ] ExXlateTests.XlateCapacities

using:

> Apple LLVM version 7.3.0 (clang-703.0.29)
>    Boost    1_59
>    Coin-Cbc 2.8.7
>    Coin-Clp 1.15.5
>    Hdf5     1.8.13-
>    Sqlite3  3.12.0
>    xml2     2.9.2
>    xml++    2.38.1
>

File: src/hdf5_back.cc.in [hunk 1]
```diff
@@ -916,7 +916,8 @@ void Hdf5Back::CreateTable(Datum* d) {
     dst_size += dst_sizes[i];
   }
 
-  const char* title = d->title().c_str();
+  std::string titlestr = d->title();
+  const char* title = titlestr.c_str();
```
Comment (gonuke) [inline src/hdf5_back.cc.in]: How does this possibly make a difference?!?!  And how did you find it?

**Comment by mbmcgarry** [comment]
This eliminates HDF5 failures on my Mac as well (Yosemite)!

**Comment by rwcarlsen** [comment]
Just to confirm @Baaaaam @mbmcgarry - this appears to also fix #981?

**Comment by Baaaaam** [comment]
yes, the only unit test error I have left is:
>[  FAILED  ] ExXlateTests.XlateCapacities

Meghan has an other one also

**Comment by Baaaaam** [comment]
On MAC OS : with hdf5 1.8.16
>   FAILED  ] 1 test, listed below:
> [  FAILED  ] ExXlateTests.XlateCapacities
> 
>  1 FAILED TEST


> Boost    1_59
> Coin-Cbc 2.9
> Coin-Clp 1.16
> Hdf5     1.8.16-
> Sqlite3  3.12.1
> xml2     2.9.2
> xml++    2.38.1

**Comment by mbmcgarry** [comment]
The total failed unit tests for me are now:
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] SymbolicFunctionTests.piecewisefunc
[  FAILED  ] ExXlateTests.XlateCapacities

**Comment by mbmcgarry** [comment]
Cyclus Core 1.3.0 (1.3.1-206-gc6fb3fe)

Dependencies:
   Boost    1_57
   Coin-Cbc 2.8
   Coin-Clp 1.15
   Hdf5     1.8.13-
   Sqlite3  3.8.10.1
   xml2     2.9.2
   xml++    <2.37

**Comment by rwcarlsen** [comment]
ping...

**Comment by Baaaaam** [comment]
works on on Mac OS and CI, @rwcarlsen start to be aggressive about this PR, I merge before he bites me! :)

**Pull request merged**